src/analyses/goto_check.cpp

```diff
@@ -1282,30 +1282,24 @@ goto_checkt::address_check(const exprt &address, const exprt &size)
     const exprt in_bounds_of_some_explicit_allocation =
       disjunction(alloc_disjuncts);
 
-    if(flags.is_unknown() || flags.is_null())
-    {
-      conditions.push_back(conditiont(
-        or_exprt(
-          in_bounds_of_some_explicit_allocation,
-          not_exprt(null_pointer(address))),
-        "pointer NULL"));
-    }
+    const bool unknown = flags.is_unknown() || flags.is_uninitialized();
 
-    if(flags.is_unknown())
+    if(unknown)
     {
       conditions.push_back(conditiont{
         not_exprt{is_invalid_pointer_exprt{address}}, "pointer invalid"});
     }
 
-    if(flags.is_uninitialized())
+    if(unknown || flags.is_null())
     {
-      conditions.push_back(
-        conditiont{or_exprt{in_bounds_of_some_explicit_allocation,
-                            not_exprt{is_invalid_pointer_exprt{address}}},
-                   "pointer uninitialized"});
+      conditions.push_back(conditiont(
+        or_exprt(
+          in_bounds_of_some_explicit_allocation,
+          not_exprt(null_pointer(address))),
```

Comment:
Not changed in this PR, but is the condition `in_bounds_of_some_explicit_allocation` redundant (or are we explicitly considering an address space where the zero pointer might point at allocated memory)?

Reply:
Good point. Yes, I think that should be redundant.